.github/workflows/linux.yml

```diff
@@ -16,187 +16,50 @@ concurrency:
-          # Should save cache only if run in the master branch of the base repo
-          # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
-          save: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
-          verbose: 2
-          key: linux-ubuntu
-          restore-keys: |
-            linux-ubuntu
-
-      - name: CMake configure - OpenVINO
-        run: |
-          cmake \
-            -G "${{ env.CMAKE_GENERATOR }}" \
-            -DENABLE_CPPLINT=OFF \
-            -DENABLE_NCC_STYLE=OFF \
-            -DENABLE_TESTS=OFF \
-            -DENABLE_STRICT_DEPENDENCIES=OFF \
-            -DCMAKE_VERBOSE_MAKEFILE=ON \
-            -DCPACK_GENERATOR=TGZ \
-            -DENABLE_JS=OFF \
-            -DENABLE_SAMPLES=OFF \
-            -DENABLE_INTEL_NPU=OFF \
-            -DENABLE_OV_ONNX_FRONTEND=OFF \
-            -DENABLE_OV_PADDLE_FRONTEND=OFF \
-            -DENABLE_OV_PYTORCH_FRONTEND=OFF \
-            -DENABLE_OV_TF_FRONTEND=ON \
-            -DENABLE_OV_TF_LITE_FRONTEND=OFF \
-            -DENABLE_INTEL_GPU=OFF \
-            -DENABLE_INTEL_NPU=OFF \
-            -DCMAKE_COMPILE_WARNING_AS_ERROR=ON \
-            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
-            -DCMAKE_C_COMPILER_LAUNCHER=ccache \
-            -DENABLE_PYTHON=ON \
-            -DENABLE_WHEEL=ON \
-            -S ${OPENVINO_REPO} \
-            -B ${BUILD_DIR}
-
-      - name: Clean ccache stats
-        run: ccache --zero-stats --show-config
-
-      - name: Cmake build - OpenVINO
-        run: cmake --build ${BUILD_DIR} --parallel --config ${{ env.CMAKE_BUILD_TYPE }}
-
-      - name: Show ccache stats
-        run: ccache --show-stats
-
-      - name: Cmake install - OpenVINO
-        run: |
-          cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}/openvino_package -P ${BUILD_DIR}/cmake_install.cmake
-          cmake -DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}/openvino_package -DCOMPONENT=python_wheels -P ${BUILD_DIR}/cmake_install.cmake
-
-      - name: Pack Artifacts
-        run: |
-          pushd ${INSTALL_DIR}
-            tar -czvf ${BUILD_DIR}/openvino_package.tar.gz *
-          popd
-
-      #
-      # Upload build artifacts and logs
-      #
-
-      - name: Upload openvino package
-        if: ${{ always() }}
-        uses: actions/upload-artifact@v4
-        with:
-          name: openvino_package
-          path: ${{ env.BUILD_DIR }}/openvino_package.tar.gz
-          if-no-files-found: 'error'
+    runs-on: aks-linux-2-cores-8gb
+    container:
+      image: 'openvinogithubactions.azurecr.io/openvino_provider:0.1.0'
+      volumes: [ "/mount:/mount" ]
+
+    steps: # TODO: revert custom PR
+    - uses: openvinotoolkit/openvino/.github/actions/openvino_provider@refs/pull/26636/head
+      id: openvino_download
+      with:
+        ov_artifact_name: ${{ env.OV_ARTIFACT_NAME }}
+        platform: 'ubuntu22_x86_64'
+        revision: 'latest_available'
+        artifacts_source: 'postcommit'
```

Comment:
does not latest_available imply post-commit?

Reply:
Not necessarily if in the future we want to support latest available RC, for example

If we want to rely solely on revision and get rid of artifacts_source, if may be problematic, since revision can be:

- latest_available (currently for post-commit only)
- HEAD (post-commit only)
- specific commit hash (post-commit only)
- rc (s3)
- specific release (s3)

Will need to create a dedicated step in an action to distinguish, e.g. commit hash from release version and choose a correct source automatically

Reply:
For now it could make this action too complicated, imho it is enough that user defines hash commit and get the latest suitable package for that (we can log its hash sum to mark the difference  if needed)

Reply:
@mryzhov, we need a way to retrieve w/o specifying a particular hash 1) latest post-commit 2) latest nightly 3) specific RC 4) specific release 5) latest available RC, if needed
We can discuss offline how it's better to provide - via a single revision or revision + source (as it's implemented now)

Reply:
1) revision = latest_available (or HEAD)
2) revision = nightly (nightly is supposed to be always latest), for specific nightly we can use revision = [2024.3.0-15549-ae01c973d53](https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/2024.3.0-15549-ae01c973d53)
3) revision = 2024.4.0rc1
4) revision = 2024.4.0
5) revision = 2024.4.0rc